content/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-pypi.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Configuring OpenID Connect in PyPI
+shortTitle: OpenID Connect in PyPI
+intro: Use OpenID Connect within your workflows to authenticate with PyPI
+versions:
+  fpt: '*'
+  ghec: '*'
+type: tutorial
+topics:
+  - Security
+  - Actions
+---
+
+{% data reusables.actions.enterprise-github-hosted-runners %}
+
+## Overview
+
+OpenID Connect (OIDC) allows your {% data variables.product.prodname_actions %} workflows to authenticate with [PyPI](https://pypi.org) to publish Python packages.
+
+This guide gives an overview of how to configure PyPI to trust {% data variables.product.prodname_dotcom %}'s OIDC as a federated identity, and demonstrates how to use this configuration in the [`pypa/gh-action-pypi-publish`](https://github.com/marketplace/actions/pypi-publish) action to publish packages to PyPI (or other Python package indices) without any manual API token management.
+
+## Prerequisites
+
+{% data reusables.actions.oidc-link-to-intro %}
+
+{% data reusables.actions.oidc-security-notice %}
+
+## Adding the identity provider to PyPI
+
+To use OIDC with PyPI, add a trust configuration that links each project on PyPI to each repository and workflow combination that's allowed to publish for it.
+
+1. Log into PyPI and navigate to the trusted publishing settings for the project you'd like to configure. For a project named `myproject`, this will be at `https://pypi.org/manage/project/myproject/settings/publishing/`.
+
+2. Configure a trust relationship between the PyPI project and a GitHub repository (and workflow within the repository). For example, if your GitHub repository is at `myorg/myproject` and your release workflow is defined in `release.yml` with an environment of `release`, you should use the following settings for your trusted publisher on PyPI.
+
+{% note %}
+
+**Note:** Enter these values correctly. Giving the incorrect user, repository, or workflow
+the ability to publish to your PyPI project is equivalent to sharing an API token.
+
+{% endnote %}
+
+- Owner: `myorg`
+- Repository name: `myproject`
+- Workflow name: `release.yml`
+- (Optionally) a GitHub Actions environment name: `release`
+
+## Updating your {% data variables.product.prodname_actions %} workflow
+
+Once your trusted publisher is registered on PyPI, you can update your release workflow to use trusted publishing.
+
+The [`pypa/gh-action-pypi-publish`](https://github.com/marketplace/actions/pypi-publish) has built-in support for trusted publishing, which can be enabled by giving its containing job the `id-token: write` permission and omitting the `username` and `password`.
+
+The following example uses `pypa/gh-action-pypi-publish` to exchange an OIDC token for a PyPI API token, which is then used to upload a package's release distributions to PyPI.
+
+```yaml{:copy}
+jobs:
+  release-build:
```

Comment:
I'd probably explicitly drop the privileges in this job or globally and emphasized that building must be done in a locked down environment that is separate from the publishing having elevated privileges.